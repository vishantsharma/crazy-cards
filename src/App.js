import React from 'react';
import { Route, Switch, HashRouter } from 'react-router-dom';
import ShowCards from '../src/components/ShowCards';
import CaptureForm from '../src/containers/CaptureForm';

const App = props => {
  return (
    <HashRouter>
    <Switch>
      <Route exact path="/" component={CaptureForm} />
      <Route exact path="/show-cards" component={ShowCards} />
    </Switch>
    </HashRouter>
)
};

export default App;
