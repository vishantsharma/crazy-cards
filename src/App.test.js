import React from 'react';
import TestRenderer from 'react-test-renderer';
import App from './App';

jest.mock('react-redux', () => {
    const ActualReactRedux = jest.requireActual('react-redux');
    return {
        ...ActualReactRedux,
        useDispatch: jest.fn().mockImplementation(() => {
            return '';
        }),
    };
});

describe('App', () => {
    it('should render', () => {
        const component = TestRenderer.create(<App />)
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})