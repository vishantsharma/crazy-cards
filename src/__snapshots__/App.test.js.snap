// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<section
  className="capture-form"
>
  <div
    className="card"
  >
    <form
      action="/formData"
      method="post"
      onSubmit={[Function]}
    >
      <div
        className="form-control"
      >
        <label
          htmlFor="title"
        >
          Title
        </label>
        <select
          className="select-form"
          id="title"
          name="title"
          onChange={[Function]}
          value=""
        >
          <option
            disabled={true}
            value=""
          >
            Please select...
          </option>
          <option
            value="mr"
          >
            Mr
          </option>
          <option
            value="mrs"
          >
            Mrs
          </option>
          <option
            value="ms"
          >
            Ms
          </option>
          <option
            value="miss"
          >
            Miss
          </option>
          <option
            value="dr"
          >
            Dr
          </option>
          <option
            value="Rev"
          >
            Rev
          </option>
          <option
            value="Other"
          >
            Other
          </option>
        </select>
      </div>
      <div
        className="form-control"
      >
        <label
          htmlFor="firstName"
        >
          First name
        </label>
        <input
          id="firstName"
          name="firstName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-control"
      >
        <label
          htmlFor="lastName"
        >
          Last name
        </label>
        <input
          id="lastName"
          name="lastName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <label
        htmlFor="dob"
      >
        Date of Birth
      </label>
      <div
        className="dob-form"
      >
        <label
          htmlFor="day"
        />
        <input
          className="dob-main"
          id="day"
          name="day"
          onChange={[Function]}
          placeholder="DD"
          type="number"
          value=""
        />
        <label
          htmlFor="month"
        />
        <input
          className="dob-main"
          id="month"
          name="month"
          onChange={[Function]}
          placeholder="MM"
          type="number"
          value=""
        />
        <label
          htmlFor="year"
        />
        <input
          className="dob-main"
          id="year"
          name="year"
          onChange={[Function]}
          placeholder="YYYY"
          type="number"
          value=""
        />
      </div>
      <div
        className="form-control"
      >
        <label
          htmlFor="houseNumber"
        >
          House Number
        </label>
        <input
          id="houseNumber"
          name="houseNumber"
          onChange={[Function]}
          type="number"
          value=""
        />
      </div>
      <div
        className="form-control"
      >
        <label
          htmlFor="postCode"
        >
          Postcode
        </label>
        <input
          id="postCode"
          name="postCode"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-control"
      >
        <label
          htmlFor="employment"
        >
          Employment Status
        </label>
        <select
          className="select-form"
          id="employment"
          name="employment"
          onChange={[Function]}
          value=""
        >
          <option
            disabled={true}
            value=""
          >
            Please select...
          </option>
          <option
            value="fulltime"
          >
            Full Time
          </option>
          <option
            value="parttime"
          >
            Part Time
          </option>
          <option
            value="selfemployed"
          >
            Self Employed
          </option>
          <option
            value="student"
          >
            Student
          </option>
          <option
            value="unemployed"
          >
            Unemployed
          </option>
          <option
            value="retired"
          >
            Retired
          </option>
          <option
            value="homemaker"
          >
            Homemaker
          </option>
        </select>
      </div>
      <div
        className="form-control"
      >
        <label
          htmlFor="income"
        >
          Annual Income
        </label>
        <input
          id="income"
          name="income"
          onChange={[Function]}
          type="number"
          value=""
        />
      </div>
      <div
        className="capture-form__actions"
      >
        <button
          type="submit"
        >
          Continue &gt;
        </button>
      </div>
    </form>
  </div>
</section>
`;
